src/main/webapp/search.js

```diff
@@ -12,12 +12,38 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-function doSearch() {
+/** Gets the topic the user searched for from the search box and redirects the page to the search-results page with a url that contains a query parameter for the topic. 
+    The user may already be on the search-results page. In that case, the page will reload with a different value for the topic query parameter. */
+function redirectToResults() {
+    return redirectToResultsHelper(document, window);
+}
 
+function redirectToResultsHelper(document, window) {
```

Comment:
It might be a good idea to add a comment to this function explaining that it exists mostly just for testing purposes with Jasmine.